### ISSUE TITLE
PresetTimeCallback

### ISSUE THREAD

**Issue by ArnoStrouwen** [SciML/ModelingToolkit.jl#3195]
It would be nice if MTK also offered [PresetTimeCallback](https://docs.sciml.ai/DiffEqDocs/stable/features/callback_functions/#PresetTimeCallback), such that we do not need to add `tstops` to the `solve` manually.

**Comment by isaacsas** [comment]
Towards that end, my most common events are discrete events where the event times are given by parameters. It would be nice to have tstops automatically calculated and setup from them instead of having to remember to pass them separately.

**Comment by ChrisRackauckas** [comment]
That is a very reasonable ask. Definitely should get implemented. Should probably wait for the overhaul of the symbolic event interface in v10 for ImplicitDiscreteSystem https://github.com/SciML/ModelingToolkit.jl/pull/3452 and be implemented in that form, but yes we should definitely add that.

**Comment by isaacsas** [comment]
@ChrisRackauckas, if there is someone that knows the new event system that would have time to look at it can we tag this as something to have ready for v10?

**Comment by ChrisRackauckas** [comment]
It's not breaking to add it, so I don't see why v10 would be delayed for it.

**Comment by ChrisRackauckas** [comment]
Actually this is already done https://docs.sciml.ai/ModelingToolkit/stable/basics/Events/#Periodic-and-preset-time-events

**Comment by isaacsas** [comment]
But that does not support symbolic parameters as the event time, or requires one to use a more general symbolic discrete callback and have to then manually pass the tstop value to solve.

**Comment by ChrisRackauckas** [comment]
I see, you want parameters in time. We can make that work, but that wasn't mentioned 😅

**Comment by isaacsas** [comment]
Yeah a way to just give a list of symbolic parameters representing preset times or representing periodic times, along with the symbolic affects, and then just have everything work via the values given in the parameter map. So avoiding any need to pass a tstop or hardcode a numerical value in the symbolic callback.

**Comment by ChrisRackauckas** [comment]
Yup that's a reasonable ask.
